supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala

```diff
@@ -164,79 +222,156 @@ final case class AwsBatchJob(jobDescriptor: BackendJobDescriptor, // WDL/CWL
         }).compile.last.map(_.get)) //if successful there is guaranteed to be a value emitted, hence we can .get this option
     }
 
-    (createDefinition[F](s"""${workflow.callable.name}-${jobDescriptor.taskCall.localName}""", uniquePath) product Kleisli.ask[F, AwsBatchAttributes]).
-      flatMap((callClient _).tupled)
+    (findOrCreateDefinition[F]() product Kleisli.ask[F, AwsBatchAttributes]).flatMap((callClient _).tupled)
+  }
+
+
+  /**
+    * Performs an md5 digest the script, checks in s3 bucket for that script, if it's not already there then persists it.
+    *
+    * @param commandLine the command line script to be executed
+    * @param scriptS3BucketName the bucket that stores the scripts
+    * @return the name of the script that was found or created
+    */
+  private def findOrCreateS3Script(commandLine :String, scriptS3BucketName: String) :String = {
```

Comment:
This surprised me - that the script needs to be uploaded as a singleton too. It seems much less likely that the script would have a collision in it, vs the rest of the task definition

Reply:
The name of the script (s3 key) is an MD5 of the content of the script so a collision should be near impossible.